### PR TITLE
k8s 1.24 rebase

### PR DESCRIPTION
**Pull request by stevekuznetsov**
go.mod: bump to logicalcluster/v2

Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

---

go.mod: bump k8s and apimachinery

Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

---

cmd: update for contextual logging feature gate

Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

---

*: adapt to k8s 1.24

Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

---

*: use logicalcluster.From()

Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

---

*: migrate setters to ClusterName --> ZZZ_DeprecatedClusterName

Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

---

clients: re-generate clients

Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

---

### CODE REVIEW

**Comment by stevekuznetsov** [comment]
e2e still needs work

File: pkg/features/kcp_features.go [hunk 1]
```diff
@@ -96,4 +96,5 @@ var defaultGenericControlPlaneFeatureGates = map[featuregate.Feature]featuregate
 	genericfeatures.ServerSideApply:         {Default: true, PreRelease: featuregate.GA},
 	genericfeatures.APIPriorityAndFairness:  {Default: true, PreRelease: featuregate.Beta},
 	genericfeatures.WarningHeaders:          {Default: true, PreRelease: featuregate.GA, LockToDefault: true}, // remove in 1.24
+	logs.ContextualLogging:                 {Default: true, PreRelease: featuregate.Alpha},
```
Comment (sttts) [inline pkg/features/kcp_features.go:97]: 🎉

**Comment by sttts** [comment]
lgtm

**Comment by stevekuznetsov** [comment]
This is going to be a fun one to keep rebased :)

File: pkg/reconciler/apis/apibinding/apibinding_reconcile_test.go [hunk 2]
```diff
@@ -956,7 +956,8 @@ func (b *bindingBuilder) Build() *apisv1alpha1.APIBinding {
 
 func (b *bindingBuilder) WithClusterName(clusterName string) *bindingBuilder {
 	//TODO (shawn-hurley): We may need to change how we set this
-	b.ClusterName = clusterName
+	// nolint:staticcheck
```
Comment (stevekuznetsov) [inline pkg/reconciler/apis/apibinding/apibinding_reconcile_test.go:959]: side-comment: it looks like this check fails only when updating a field that's deprecated, not when creating a struct literal and setting it ...

File: docs/developers/controllers.md [hunk 3]
```diff
@@ -10,12 +10,12 @@ Here are what keys look like for an object `foo` for both cluster-scoped and nam
...
-|root|my-org|root:my-org|-|root:my-org#$#foo|
-|root|my-org|root:my-org|default|default/root:my-org#$#foo|
-|my-org|my-workspace|my-org:my-workspace|-|my-org:my-workspace#$#foo|
-|my-org|my-workspace|my-org:my-workspace|default|default/my-org:my-workspace#$#foo|
+|-|-|root|-|root|foo|
+|-|-|root|default|default/root|foo|
+|root|my-org|root:my-org|-|root:my-org|foo|
+|root|my-org|root:my-org|default|default/root:my-org|foo|
+|my-org|my-workspace|my-org:my-workspace|-|my-org:my-workspace|foo|
+|my-org|my-workspace|my-org:my-workspace|default|default/my-org:my-workspace|foo|
```
Comment (sttts) [inline docs/developers/controllers.md:18]: this is totally unreadable now, isn't it? :D

**Comment by sttts** [comment]
/approve
Lgtm (after updating go.mod).

File: docs/developers/controllers.md (hunk 3 above)
Comment (stevekuznetsov) [reply docs/developers/controllers.md:18]: eh, // TODO :)

**Comment by stevekuznetsov** [comment]
https://github.com/kcp-dev/kcp/issues/1694

**Pull request merged**